pkg/registry/apis/query/query.go

```diff
@@ -413,6 +427,44 @@ func (b *QueryAPIBuilder) handleExpressions(ctx context.Context, req parsedReque
 	return qdr, nil
 }
 
+// copied from pkg/expr/nodes.go from within the Execute method
+func handleSqlInput(ctx context.Context, dr *backend.DataResponse) (mathexp.Results, error) {
```

Comment:
If we put a call to this func in the `Convert` method that is exported from the SSE service, I think that would get rid of the duplication, and the need to call Convert different or not. Will just need to add an argument to Convert that is something like `forSQL bool`

Reply:
this sounds good but i think would mean editing both code paths at the same time which is not tricky to do but is kind of tricky to feel confident in maybe? I wonder if it might make sense to do as a follow up pr?

Reply:
I'm good with either. Even though Convert is a hot spot, since this is under a flag, I think the change should be safe as long as we don't accidentally invert the logic ;-)